src/librustc/ty/mod.rs

```diff
@@ -2746,12 +2746,24 @@ fn associated_item_def_ids<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>,
     Rc::new(vec)
 }
 
+fn describe_def<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>,
+                          def_id: DefId)
+                          -> Option<Def> {
+   let id = tcx.hir.as_local_node_id(def_id).unwrap();
+   let item = tcx.hir.expect_item(id);
+   match item.node{
+       hir::ItemTrait(..) => Some(Def::Trait(def_id)),
+       _ => None
```

Comment:
There are way more `Def` variants than that ;). Anyway, see https://github.com/rust-lang/rust/pull/41534#discussion_r113594211